### PR TITLE
1.1.1: Handle trailing zeros added by BINARY type column

### PR DESCRIPTION
**Pull request by Samira-El**

### CODE REVIEW

File: tap_mysql/sync_strategies/common.py [hunk 1]
```diff
@@ -100,8 +100,22 @@ def row_to_singer_record(catalog_entry, version, row, columns, time_extracted):
...
+            #
+            # ref: https://dev.mysql.com/doc/refman/8.0/en/binary-varbinary.html
+            #
+            # Leaving the trailing zeroes makes FT & Incremental behave differently than Binlog and we end up with
+            # inconsistent state records.
+            #
+            # Example: for a binary value b'pk0'
+            #   - Binlog would read it as b'pk0' then in HEX it would be b'706b30'
+            #   - FT & Inc would read it as b'pk0\x00\x00\x00\x00....\x00' then in HEX it would be b'706b300000...0'
+            stripped_utf8_elem = elem.decode().rstrip('\x00').encode()
```
Comment (ivan-transferwise) [inline tap_mysql/sync_strategies/common.py]: That makes sense. Just several concerns:
1. `decode()` and `encode()` take `utf-8` as default encoding. Wouldn't it be a problem?
2. Can applying `rstrip()` cause false positives? For example, is it possible that you will strip unnecessary zeros?
3. Also, there is no test for it, would be good to have small example to see an impact of it?

Comment (Samira-El) [reply tap_mysql/sync_strategies/common.py]: Thank you, very valid concerns:

1. I left the default encoding as is on purpose, we don't what type of encoding will the data be using, so I left the standard one.
2.  I have tested inserting binary data that ends with 0s and it worked fine :).
3. This is only observable in two cases: 
    - when using INCREMENTAL then switching to BINLOG (vice versa) which usually doesn't happen. 
   - in Pipelinewise when we initially sync using FastSync(alt FullTable) then binlog.

I have added an integration test in PPW that syncs a table that has Binary & varbinary column using first FastSync then binlog.

File: tap_mysql/sync_strategies/common.py [hunk 2]
```diff
@@ -100,8 +100,22 @@ def row_to_singer_record(catalog_entry, version, row, columns, time_extracted):
...
+
+            # encode the stripped elem to hex
+            row_to_persist += (codecs.encode(stripped_utf8_elem, 'hex'),)
```
Comment (koszti) [inline tap_mysql/sync_strategies/common.py]: this is not in sync with PLW fastsync [here](url) . Is that fine not decoding to utf-8 like we do it in fastsync?

Comment (Samira-El) [reply tap_mysql/sync_strategies/common.py]: It is in sync, just no explicitly, `json.dump` later called by singer's `write_record` function does decode bytes to utf8.

Comment (koszti) [reply tap_mysql/sync_strategies/common.py]: ok cool, thank you

**Pull request merged**